frontend/src/js/components/common/list.tsx

```diff
@@ -0,0 +1,424 @@
+    },
+    ['&.resizing']: {
+      background: isDarkMode(theme.palette.mode) ? theme.palette.grey[200] : theme.palette.grey[900]
+    }
+  }
+}));
+
+export const minCellWidth = 150;
+
+export const calculateResizeChange = ({ columnElements, columnHeaders, e, index, prev, selectable }) => {
+  const isShrinkage = prev > e.clientX ? -1 : 1;
+  const columnDelta = Math.abs(e.clientX - prev) * isShrinkage;
+  const relevantColumns = getRelevantColumns(columnElements, selectable);
+  const canModifyNextColumn = index + 1 < columnHeaders.length - 1;
+
+  return relevantColumns.reduce((accu, element, columnIndex) => {
+    const currentWidth = element.offsetWidth;
+    const column = { attribute: columnHeaders[columnIndex + 1].attribute, size: currentWidth };
+    if (canModifyNextColumn && index === columnIndex) {
+      column.size = currentWidth + columnDelta;
+    } else if (canModifyNextColumn && index + 1 === columnIndex) {
+      column.size = currentWidth - columnDelta;
+    }
+    accu.push(column);
+    return accu;
+  }, []);
+};
+const getRelevantColumns = (columnElements, selectable) => [...columnElements].slice(selectable ? 1 : 0, columnElements.length - 1);
+const getTemplateColumns = (columns, selectable) =>
+  selectable ? `52px ${columns} minmax(${minCellWidth}px, 1fr)` : `${columns} minmax(${minCellWidth}px, 1fr)`;
+
+const getColumnsStyle = (columns, defaultSize, selectable) => {
+  const template = columns.map(({ size }) => `minmax(${minCellWidth}px, ${size ? `${size}px` : defaultSize})`);
+  // applying styles via state changes would lead to less smooth changes, so we set the style directly on the components
+  return getTemplateColumns(template.join(' '), selectable);
+};
+export const CommonList = <T extends wID>(props: CommonListProps<T>) => {
+  const {
+    columnHeaders,
+    customColumnSizes = [],
+    listItems,
+    listState,
+    idAttribute,
+    onChangeRowsPerPage,
+    PaginationProps = {},
+    onExpandClick,
+    onResizeColumns,
+    onPageChange,
+    onSelect,
+    onSort,
+    pageLoading,
+    sortingNotes,
+    ListItemComponent
+  } = props;
+  const { page: pageNo = defaultPage, perPage: pageLength = defaultPerPage, selection: selectedRows = [], sort = {}, total: pageTotal = 1 } = listState;
+  const { direction: sortDown = SORTING_OPTIONS.desc, key: sortCol } = sort;
+  const listRef = useRef<HTMLDivElement | null>(null);
+  const selectedRowsRef = useRef(selectedRows);
+  const initRef = useRef<number | null>(null);
+  const [resizeTrigger, setResizeTrigger] = useState(false);
+
+  const size = useWindowSize();
+  const selectable = !!onSelect;
+  const { classes } = useStyles();
+
+  useEffect(() => {
+    selectedRowsRef.current = selectedRows;
+  }, [selectedRows]);
+
+  useEffect(() => {
+    if (!listRef.current) {
+      return;
+    }
+    const relevantColumns = getRelevantColumns(listRef.current?.querySelector('.deviceListRow')?.children, selectable);
```

Comment:
more of a note to self: the `.deviceListRow` should probably be renamed at some point... but OTOH it could probably also be changed to be a subgrid and then this part could maybe be changed...